spec/controllers/migrations/external_controller_spec.rb

```diff
@@ -83,6 +83,250 @@
     end
   end
 
+  describe '#create_user' do
+    let(:valid_user_params) do
+      role = create(:role, provider: 'greenlight')
+      {
+        name: 'user',
+        email: 'user@users.com',
+        language: 'language',
+        role: role.name
+      }
+    end
+
+    before { clear_enqueued_jobs }
+
+    describe '#generate_secure_pwd' do
+      before { allow_any_instance_of(described_class).to receive(:generate_secure_pwd).and_call_original }
+
+      it 'generates a secure random complex password' do
+        pwd_pattern = %r{\A(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[`@%~!#£$\\^&*()\]\[+={}/|:;"'<>\-,.?_ ]).{8,}\z}
```

Comment:
We can make a small refactoring on the user model creating a `PWD_PATTERN` that holds the pattern as a single source of truth and configuration,
The user model password format validation itself can use the constant,
The test and any other area can refer to it,